Sources/Basics/FileSystem/FileSystem+Extensions.swift

```diff
@@ -270,6 +276,20 @@ extension FileSystem {
     }
 }
 
+extension FileSystem {
+    
+    public func getOrCreateSwiftPMInstalledBinariesDirectory() throws -> AbsolutePath {
+        let binDir = try self.dotSwiftPMInstalledBinsDir
+        
+        // create if necessary
+        if !self.exists(binDir) {
+            try self.createDirectory(binDir, recursive: true)
+        }
+        
+        return binDir
```

Comment:
It's better to fully follow the pattern that other directories use as that accounts for symlinking to `~/Library` and also concurrent access:

```suggestion
        let idiomaticInstalledBinariesDirectory = try self.swiftPMInstalledBinariesDirectory
        // Create idiomatic if necessary
        if !self.exists(idiomaticInstalledBinariesDirectory) {
            try self.createDirectory(idiomaticInstalledBinariesDirectory, recursive: true)
        }
        // Create ~/.swiftpm if necessary
        if !self.exists(try self.dotSwiftPM) {
            try self.createDirectory(self.dotSwiftPM, recursive: true)
        }
        // Create ~/.swiftpm/bin symlink if necessary
        // locking ~/.swiftpm to protect from concurrent access
        try self.withLock(on: self.dotSwiftPM, type: .exclusive) {
            if !self.exists(try self.dotSwiftPMInstalledBinariesDirectory, followSymlink: false) {
                try self.createSymbolicLink(
                    self.dotSwiftPMInstalledBinariesDirectory,
                    pointingAt: idiomaticInstalledBinariesDirectory,
                    relative: false
                )
            }
        }
        return idiomaticInstalledBinariesDirectory
```